livre/eq/Config.cpp

```diff
@@ -94,14 +100,27 @@ class Config::Impl
         , eventMapper( EventHandlerFactoryPtr( new EqEventHandlerFactory ))
         , redraw( true )
         , dataFrameRange( INVALID_FRAME_RANGE )
+        , dataSourceRange( std::numeric_limits<uint8_t>::min(), // This will change with the new TF
+                           std::numeric_limits<uint8_t>::max( ))
     {}
 
     void gatherHistogram( const Histogram& histogram, const float area, const uint32_t currentId )
     {
-      // If we get a very old frame skip it
+        // If we get a very old frame skip it
         if( !histogramQueue.empty() && currentId < histogramQueue.back().id )
             return;
 
+        // Always enlarge the range
+        dataSourceRange[ 0 ] =
+                histogram.getMin() < dataSourceRange[ 0 ] ? histogram.getMin() :
+                                                            dataSourceRange[ 0 ];
+
+        dataSourceRange[ 1 ] =
+                histogram.getMax() > dataSourceRange[ 1 ] ? histogram.getMax() :
+                                                            dataSourceRange[ 1 ];
+
+        config->getFrameData().getVolumeSettings().setDataSourceRange( dataSourceRange );
+
         const ViewHistogram viewHistogram( histogram, area, currentId );
         auto it = histogramQueue.begin();
         while( it != histogramQueue.end( ))
```

Comment:
Use a for? You are always increasing it at the end of the iteration except if you break or return.